aiohttp/helpers.py

```diff
@@ -852,6 +853,15 @@ def set_cookie(
         self._cookies[name] = value
         c = self._cookies[name]
 
+        if DEBUG:
+            cookie_length = (
+                len(name) + len(value) + sum(len(k) + len(v) for k, v in c.items())
```

Comment:
Have you reproduced the issue in Chrome and tested the different lengths? I think I saw somewhere recently that the actual limit is on the total length of the string that is output in the header. If so, this probably misses some spaces or joining characters.

Reply:
Nice note, what about characters that are more than 1 byte like ÿ that is as far as I know stored as %C3%BF which is 6 bytes. doing `len('ÿ')` results into 1.

Reply:
That is a good point, maybe they should be converted to `bytes`?

Reply:
Consider using `sys.getsizeof()` instead.

Reply:
Note that for complex structures you'd have to go for recursion that may be expensive: https://code.activestate.com/recipes/577504/.

Reply:
I think that won't be a problem because we are computing the length of the raw cookie header which is a string.

Reply:
A length of a Unicode string may be smaller than the length of its underlying byte representation since some char would be represented as more than one byte:
```pycon
$ ipython
Python 3.9.0 (default, Oct 26 2020, 13:08:59) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.19.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: len('ї')
Out[1]: 1

In [2]: len('ї'.encode())
Out[2]: 2
```

Reply:
Or are you implying that only ASCII is allowed there? Is it actually checked anywhere?

Reply:
He was responding to your comment about needing recursion, which shouldn't be needed as they are just strings.

We'll need to ensure we are checking the size of the bytes and not the string length as mentioned previously.

This thread is also referencing old code, so I'll resolve it to avoid confusion. We'll make sure that a 2-byte character is included in the tests to cover this case.